File: src/main/java/com/dec/decisland/item/CustomItemProperties.java
package com.dec.decisland.item;

public class CustomItemProperties {
    public int burnTime;
    public float compostableChance;
    private CustomItemProperties(Builder builder) {
        this.burnTime = builder.burnTime;
        this.compostableChance = builder.compostableChance;
    }

    public static class Builder {
        private int burnTime = 0;
        private float compostableChance = 0;
        public Builder() {

        }
        public Builder burnTime(int burnTime) {
            this.burnTime = burnTime;
            return this;
        }
        public Builder compostableChance(float compostableChance) {
            this.compostableChance = compostableChance;
            return this;
        }
        public CustomItemProperties build() {
            return new CustomItemProperties(this);
        }
    }
}

File: src/main/java/com/dec/decisland/item/CustomItemProperties.kt
package com.dec.decisland.item

class CustomItemProperties private constructor(builder: Builder) {
    data class CooldownBonus(
        val category: ItemConfig.WeaponCooldownCategory,
        val ticksPerFourTicks: Float,
    )

    @JvmField
    val burnTime: Int = builder.burnTime

    @JvmField
    val compostableChance: Float = builder.compostableChance

    @JvmField
    val accessoryCooldownBonus: CooldownBonus? = builder.accessoryCooldownBonus

    class Builder {
        internal var burnTime: Int = 0
        internal var compostableChance: Float = 0.0f
        internal var accessoryCooldownBonus: CooldownBonus? = null

        fun burnTime(burnTime: Int): Builder = apply {
            this.burnTime = burnTime
        }

        fun compostableChance(compostableChance: Float): Builder = apply {
            this.compostableChance = compostableChance
        }

        fun accessoryCooldownBonus(accessoryCooldownBonus: CooldownBonus?): Builder = apply {
            this.accessoryCooldownBonus = accessoryCooldownBonus
        }

        fun build(): CustomItemProperties = CustomItemProperties(this)
    }
}
